apps/www/pages/api/prolitteris.ts

```diff
@@ -0,0 +1,125 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import truncateIP from '../../lib/api/TruncateIP'
+import withReqMethodGuard from '../../lib/api/withReqMethodGuard'
+import HTTPMethods from '../../lib/api/HTTPMethods'
+import crypto from 'node:crypto'
+
+const {
+  PROLITTERIS_MEMBER_ID,
+  PROLITTERIS_DOMAIN,
+  PROLITTERIS_USER_AGENT,
+  PUBLIC_BASE_URL,
+  PROLITTERIS_DEV_UID,
+  PROLITTERIS_HASH_SECRET = '',
+} = process.env
+
+/**
+ * Generate a sha256 hash from a string, number, object or array
+ * @param input
+ * @returns
+ */
+function getHash(input: string | number | object): string {
+  const hash = crypto.createHash('sha256')
+  if (input instanceof Object || Array.isArray(input)) {
+    hash.update(JSON.stringify(input))
+  } else {
+    hash.update(input.toString())
+  }
+  hash.update(PROLITTERIS_HASH_SECRET)
+  return hash.digest('hex')
+}
+
+/**
+ *
+ * @param req
+ * @param res
+ */
+async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const requestIps =
+    request.headers['x-forwarded-for'] || request.socket.remoteAddress
+
+  const ua = request.headers['user-agent']
+
+  // throw error if no IP is supplied
+  if (!requestIps) {
+    throw new Error('IP undefined')
+  }
+
+  // if x-forwarded-for contains an array of ip's, use the left most (client)
+  const requestIp = Array.isArray(requestIps) ? requestIps[0] : requestIps
+
+  // Query Parameters of request
+  // 1) paid (string, 'pw' || 'na'): request by paying user (pw) or public (na)
+  // 2) uid (string): repoId of the article
+  // 3) path (string): article slug
+
+  const { paid, uid, path } = request.query
+
+  // Check that all query parameters are defined.
+  if (!paid) {
+    return response.status(400).json({
+      body: 'paid parameter required.',
+    })
+  }
+
+  if (paid !== 'na' && paid !== 'pw') {
+    return response.status(400).json({
+      body: "Paid parameter must be string 'na' or 'pw'",
+    })
+  }
+
+  if (!uid) {
+    return response.status(400).json({
+      body: 'uid parameter required.',
+    })
+  }
+
+  if (!path) {
+    return response.status(400).json({
+      body: 'path parameter required.',
+    })
+  }
+
+  // create unique C-Parameter for each request (20 characters hex) from the ip and user agent
+  const cParam: string = getHash([requestIp, ua]).substring(0, 20)
+  // replace repoID forward slash with dash, as forward slashes are not allowed in uid parameter
+  const uidParam = PROLITTERIS_DEV_UID
+    ? PROLITTERIS_DEV_UID
+    : Array.isArray(uid)
```

Comment:
When could `uid` become an array?

Reply:
never, but typescript complains and I can't change the predefined next types 🤨

Reply:
made a different suggestion for the issue